.github/workflows/rust.yml

```diff
@@ -45,11 +45,16 @@ jobs:
 
       - name: Configure cache
         uses: actions/cache@704facf57e6136b1bc63b828d79edcd491f0ee84 # @v3.3.2
+        env:
+          cache-name: cargo-fmt
         with:
           path: |
-            ~/.cargo/registry
-            ~/.cargo/git
+            ${{ runner.os == 'Windows' && format('{0}/.cargo/registry', env.USERPROFILE) || '~/.cargo/registry' }}
+            ${{ runner.os == 'Windows' && format('{0}/.cargo/git', env.USERPROFILE) || '~/.cargo/git' }}
```

Comment:
Are you really sure we need all this mess? I'm suspecting it worked fine the way it was.